File: TeamCode/src/main/java/com/info1robotics/bobot/tasks/LinkageTask.java
package com.info1robotics.bobot.tasks;

public class LinkageTask extends Task{
    public enum Direction {
        EXTENDED(1.0),
        LOWERED(.00),
        MANUAL(.5);
        public final double posLeft;
        public final double posRight;
        Direction(double pos) {
            this.posLeft = pos;
            this.posRight = 1.0-pos;
        }
    }
    Direction direction=Direction.MANUAL;
    double position=this.direction.posLeft;
    double manualPos=.5;
    double posLeft;
    double posRight;
    @Override
    public void run() {
        if(direction==Direction.MANUAL)
        {
            posLeft=manualPos;
            posRight=1.0-manualPos;
        }
        else
        {
            posLeft=direction.posLeft;
            posRight=direction.posRight;
        }
//        context.linkageLeft.setPosition(posLeft);
//        context.linkageRight.setPosition(posRight);
        state=State.FINISHED;
    }
    LinkageTask(double position)
    {
        this.manualPos=position;
    }
    LinkageTask(Direction direction)
    {
        this.direction=direction;
    }

}

File: TeamCode/src/main/kotlin/com/info1robotics/bobot/tasks/LinkageTask.kt
package com.info1robotics.bobot.tasks

import androidx.core.math.MathUtils.clamp
import com.qualcomm.robotcore.hardware.DcMotorSimple

class LinkageTask( var power:Double,val direction:Direction):Task() {
    enum class Direction
    {
        EXTEND,
        LOWER,
        NONE
    }

    val speed = power
    override fun run() {
        if(direction == Direction.LOWER)
        { context.sliderServo.direction = DcMotorSimple.Direction.FORWARD
        context.sliderServoLeft.direction= DcMotorSimple.Direction.REVERSE}
        else if(direction == Direction.EXTEND)
        {   context.sliderServo.direction = DcMotorSimple.Direction.REVERSE
            context.sliderServoLeft.direction= DcMotorSimple.Direction.FORWARD}
        if(direction != Direction.NONE) {
            context.sliderServo.power = speed
            context.sliderServoLeft.power = speed
        } else {
            context.sliderServo.power = 0.0
            context.sliderServoLeft.power = 0.0
        }

        println("Slider servo run")
    }


}